.github/workflows/lint.yaml

```
@@ -0,0 +1,47 @@
+name: Lint
+on:
+  pull_request:
+    paths-ignore:
+      - README.md
+    branches:
+      - 'main'
+      - 'develop'
+      - 'feature/**'
+
+env:
+  CARGO_INCREMENTAL: 0
+  RUSTFLAGS: "-Dwarnings"
+
+jobs:
+  rustfmt:
+    runs-on: [self-hosted, Linux, small]
+    container: mobilecoin/rust-sgx-base:latest
+    strategy:
+      matrix: { dir: ['core', 'trusted', 'untrusted'] }
```

Nit: Let's use indentation rather than braces
```suggestion
      matrix:
        dir: ['core', 'trusted', 'untrusted']
```